File: Roadnik/Data/TimedStorageEntry.cs
using Ax.Fw.Storage.Data;

namespace Roadnik.Data;

internal record TimedStorageEntry(
  long UnixTimeMs,
  string RoomId,
  string Username,
  float Latitude,
  float Longitude,
  float Altitude,
  float? Speed = null,
  float? Accuracy = null,
  float? Battery = null,
  float? GsmSignal = null,
  float? Bearing = null)
{
  public static TimedStorageEntry FromStorageEntry(DocumentEntry<StorageEntry> _document)
  {
    var data = _document.Data;
    return new TimedStorageEntry(
      _document.Created.ToUnixTimeMilliseconds(),
      data.RoomId,
      data.Username,
      data.Latitude,
      data.Longitude,
      data.Altitude,
      data.Speed,
      data.Accuracy,
      data.Battery,
      data.GsmSignal,
      data.Bearing);
  }
}

File: www-vue/src/data/backend/TimedStorageEntry.ts
import type { AppId } from "../Guid";

export interface TimedStorageEntry {
  AppId: AppId;
  UnixTimeMs: number;
  Username: string;
  Latitude: number;
  Longitude: number;
  Altitude: number;
  Speed?: number | undefined | null;
  Accuracy?: number | undefined | null;
  Battery?: number | undefined | null;
  GsmSignal?: number | undefined | null;
  Bearing?: number | undefined | null;
  HR?: number | null;
}
